core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -3220,6 +3179,187 @@ private CompiledQuery sqlTableBackup(SqlExecutionContext executionContext) throw
         }
     }
 
+    private class DropStatementCompiler implements Closeable {
+        private final CharSequenceObjHashMap<String> dropTablesFailedList = new CharSequenceObjHashMap<>();
+        private final ObjHashSet<TableToken> dropTablesList = new ObjHashSet<>();
+
+        @Override
+        public void close() {
+            dropTablesList.clear();
+            dropTablesFailedList.clear();
+        }
+
+        private CompiledQuery dropDatabase(SqlExecutionContext executionContext) {
+            // collect table names
+            dropTablesList.clear();
+            engine.getTableTokens(dropTablesList, false);
+            if (dropTablesList.size() == 0) {
+                LOG.info().$("nothing to delete").$();
+                return compiledQuery.ofDrop();
+            }
+            greedyDropTablesList(executionContext.getSecurityContext());
+            return compiledQuery.ofDrop();
+        }
+
+        private CompiledQuery dropExecutorSelector(SqlExecutionContext executionContext) throws SqlException {
+            // expected syntax, one of:
+            //   - DROP TABLE [ IF EXISTS ] name [;]  in this case we call 'dropTable'
+            //   - DROP TABLES name(, name)* [;]      in this case we call 'dropMultipleTables'
```

Comment:
I think DROP DATABASE could be confusing because :
- we've only one database and we can't drop it 
- command actually removes tables and keeps 'database' (it's still possible to connect to it)
- implementation is logically inconsistent with PG's CREATE/DROP DATABASE 

As such I think it'd be better to use different syntax, e.g. DROP ALL TABLES .
What do you think ?

Reply:
`DROP ALL TABLES` it is, however we do not relete ALL TABLES as system tables are kept.